src/redisearch_api.c

```diff
@@ -45,19 +47,18 @@ IndexSpec* RediSearch_CreateIndex(const char* name, const RSIndexOptions* option
     spec->stopwords = NewStopWordListCStr((const char **)options->stopwords,
                                                          options->stopwordsLen);
   }
+
   return spec;
 }
 
 void RediSearch_DropIndex(IndexSpec* sp) {
-  RWLOCK_ACQUIRE_WRITE();
```

Comment:
If you do not get the lock on Drop the GC might be in the middle of garbage collection and it will crash ...